sql/core/src/test/scala/org/apache/spark/sql/execution/PlannerSuite.scala

```diff
@@ -474,9 +475,9 @@ class PlannerSuite extends SharedSparkSession with AdaptiveSparkPlanHelper {
       Inner,
       None,
       shuffle,
-      shuffle)
+      shuffle.copy())
```

Comment:
If this were the same as the other child of SortMergeJoinExec then no reuse would be required.

Reply:
I don't follow this part, could you please let me know how copying here makes it different?

Reply:
Yes, so I think the whole point of exchange reuse is to find different exchange instances which result the same data and keep only one instance and reuse its output where we can (i.e. call `execute`/`executeBroadcast` multiple times on one exchange instance instead of call `execute`/`executeBroadcast` once on each instances). In this example the 2 children points to the same exchange instance (`shuffle`) so there is no point in reuse here.
(We could wrap one of them in a `ReusedExcahngeExec` node, but it wouldn't make any difference from performance point of view.)

Reply:
Makes sense, although I think the reuse exchange logic that we have doesn't check if the instances are the same, and will replace it with a reuse exchange anyway, but good to have a different instance here.